accumulators/include/alps/accumulators/feature/count.hpp

```diff
@@ -91,82 +87,64 @@ namespace alps {
                         os << " #" << alps::short_print(count());
                     }
 
-                    void save(hdf5::archive & ar) const {
-                        if (m_count==0) {
-                            throw std::logic_error("Attempt to save an empty result" + ALPS_STACKTRACE);
-                        }
-                        ar["count"] = m_count;
-                    }
-
-                    void load(hdf5::archive & ar) {
-                        count_type cnt;
-                        ar["count"] >> cnt;
-                        if (cnt==0) {
-                            throw std::runtime_error("Malformed archive containing an empty result"
-                                                     + ALPS_STACKTRACE);
-                        }
-                        m_count=cnt;
-                    }
+                    void save(hdf5::archive & ar) const;
+                    void load(hdf5::archive & ar);
 
                     static std::size_t rank() { return 1; }
-                    static bool can_load(hdf5::archive & ar) { // TODO: make archive const
-                        return ar.is_data("count");
-                    }
+                    static bool can_load(hdf5::archive & ar);
 
                     template<typename U> void operator+=(U const & arg) { augadd(arg); }
                     template<typename U> void operator-=(U const & arg) { augsub(arg); }
                     template<typename U> void operator*=(U const & arg) { augmul(arg); }
                     template<typename U> void operator/=(U const & arg) { augdiv(arg); }
 
-                    inline void reset() {
-                        throw std::runtime_error("A result cannot be reseted" + ALPS_STACKTRACE);
-                    }
+                    void reset();
 
                 private:
 
                     // TODO: make macro ...
-                    template<typename U> void augadd(U const & arg, typename boost::enable_if<boost::is_scalar<U>, int>::type = 0) {
+                    template<typename U> void augadd(U const & arg, typename std::enable_if<std::is_scalar<U>::value, int>::type = 0) {
                         if (m_count == 0)
                             throw std::runtime_error("The results needs measurements" + ALPS_STACKTRACE);
                         B::operator+=(arg);
                     }
-                    template<typename U> void augadd(U const & arg, typename boost::disable_if<boost::is_scalar<U>, int>::type = 0) {
+                    template<typename U> void augadd(U const & arg, typename std::enable_if<!std::is_scalar<U>::value, int>::type = 0) {
                         if (m_count == 0 || arg.count() == 0)
                             throw std::runtime_error("Both results needs measurements" + ALPS_STACKTRACE);
```

Comment:
need, not needs (and yes, I know it's not your mistake... just happened to find it)